### ISSUE TITLE
Redesign the grass texture

### ISSUE THREAD

**Issue by IntegratedQuantum** [PixelGuys/Cubyz#190]
Currently the grass textures shows some dirt on the sides(top row). This might be good for digging up the terrain, but it's pretty bad for natural terrain:
![Screenshot at 2023-11-23 15-34-35](https://github.com/PixelGuys/Cubyz/assets/43880493/7386870f-b9b4-4c2a-adc7-0bd3ac674f7f)
One solution would be to use a full side texture(bottom row). However that does look pretty bad when on top of a regular dirt block.

Is there a way to dynamically decide between the two? Maybe using a rotation mode?

**Comment by careeoki** [comment]
> Is there a way to dynamically decide between the two? Maybe using a rotation mode?
> 
> Some kind of connected textures approach could help here.

@ikabod-kee No, this is something we still need. Grass should use a specific side texture if there isn't another grass block to connect to

**Comment by careeoki** [comment]
> Grass should use a specific side texture 

Okay, I've changed my mind on this because of recent block additions+work on biomes.

Instead of having unique side textures, grass blocks should have the grass texture on all sides, and the transition can be made with #889 

benefits:
- Grass blocks can transition to any other block. This allows biomes to control their colour palette more. example: a biome using dirt instead of soil won't have a stripe of soil on all of the grass blocks
![image](https://github.com/user-attachments/assets/e5d7bc2c-dc4d-4b8b-b4cb-d4643f6b4d0d)
- The player can build with a 6-sided grass block (which is super useful)

**Comment by IntegratedQuantum** [comment]
Wouldn't a better solution just be to use the dirt grass, or just put soil under all the regular grass blocks?

**Comment by codemob-dev** [comment]
Could this be done with a rotation mode similar to how ores can be overlayed on other blocks?

**Comment by IntegratedQuantum** [comment]
Yeah, we could make grass an ore, but then we might not have enough information to distinguish between connecting to a grass layer below or not.
